pkg/kritis/metadata/metadata.go

```diff
@@ -130,49 +109,43 @@ func GetVulnerabilityFromOccurrence(occ *grafeas.Occurrence) *Vulnerability {
 	return &vulnerability
 }
 
-func GetRawAttestationsFromOccurrence(occ *grafeas.Occurrence) ([]RawAttestation, error) {
-	ras := []RawAttestation{}
-	att := occ.GetAttestation().GetAttestation()
-	switch att.Signature.(type) {
+// GetAttestationsFromOccurrence parses Attestations from PgpSignedAttestation
+// and GenericSignedAttestation Occurrences. A PgpSignedAttestation has one
+// signature and is parsed into one Attestation. A GenericSignedAttestation may
+// have multiple signatures, which are parsed into multiple Attestations.
+func GetAttestationsFromOccurrence(occ *grafeas.Occurrence) ([]cryptolib.Attestation, error) {
+	atts := []cryptolib.Attestation{}
+	occAtt := occ.GetAttestation().GetAttestation()
+	switch occAtt.Signature.(type) {
 	case *attestationpb.Attestation_PgpSignedAttestation:
-		psa := att.GetPgpSignedAttestation()
-		ra := RawAttestation{
-			SignatureType: PgpSignatureType,
-			Signature: RawSignature{
-				PublicKeyId: psa.GetPgpKeyId(),
-				Signature:   psa.GetSignature(),
-			},
-			SerializedPayload: []byte{},
+		psa := occAtt.GetPgpSignedAttestation()
+		att := cryptolib.Attestation{
+			PublicKeyID: psa.GetPgpKeyId(),
+			Signature:   []byte(psa.GetSignature()),
 		}
-		ras = append(ras, ra)
+		atts = append(atts, att)
 	case *attestationpb.Attestation_GenericSignedAttestation:
-		gsa := att.GetGenericSignedAttestation()
+		gsa := occAtt.GetGenericSignedAttestation()
 		for _, sig := range gsa.GetSignatures() {
-			newSig := RawSignature{
-				PublicKeyId: sig.PublicKeyId,
-				Signature:   string(sig.Signature),
-			}
-			ra := RawAttestation{
-				SignatureType:     GenericSignatureType,
-				Signature:         newSig,
+			att := cryptolib.Attestation{
+				PublicKeyID:       string(sig.PublicKeyId),
+				Signature:         sig.Signature,
 				SerializedPayload: gsa.GetSerializedPayload(),
 			}
-			ras = append(ras, ra)
+			atts = append(atts, att)
 		}
 	default:
-		return nil, fmt.Errorf("Unknown signature type for attestation %v", att)
+		return nil, fmt.Errorf("Unknown signature type for attestation %v", occAtt)
 	}
-	return ras, nil
+	return atts, nil
 }
 
-// For testing purposes. Should not be used as part of metadata external API.
-func MakeRawAttestation(sigType SignatureType, sig, id, payload string) RawAttestation {
-	return RawAttestation{
-		SignatureType: sigType,
-		Signature: RawSignature{
-			Signature:   sig,
-			PublicKeyId: id,
-		},
-		SerializedPayload: []byte(payload),
+// MakeAttestation is testing purposes. Should not be used as part of metadata
+// external API.
+func MakeAttestation(publicKeyID string, signature string, serializedPayload []byte) cryptolib.Attestation {
```

Comment:
Not added in this CL, but think this kind of helper function is generally more harmful than beneficial. It's not that hard to directly construct cryptolib.Attestation.

Alternatively, move it to *_test.go and rename it to MakeTestAttestation. It's better to duplicate this function in multiple test files than to have iit here with the production code.

Reply:
Done, removed